techhive/server.js

```diff
@@ -13,18 +14,36 @@ import { SavedInternship } from './models/savedInternships.js';
 import {router as authRoutes} from './routes/authRoutes.js';
 import { authenticateToken } from './routes/authRoutes.js';
 import sgMail from '@sendgrid/mail';
-sgMail.setApiKey('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');
+import dotenv from 'dotenv';
+import messageRoutes from './routes/messages.js'
+
+sgMail.setApiKey('//Took API key out to prevent privacy exposure in Github');
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(__filename);
 
 const app = express();
-
+dotenv.config();
 
 app.use(cors());
 app.use(express.json()); //Middleware for parsing JSON bodies from HTTP requests
 app.use(morgan('combined'));
 
+//MongoDb Database Setup
+mongoose.connect(process.env.MONGO_URL, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+ })
+.then(() => {
+  console.log("DB Connection Successfull")
+ }).catch((err) => {
+  console.log(err.message)
+ })
+
+const server = app.listen(process.env.PORT, ()=> {
```

Comment:
Can you explain what this line of code is doing? Is it specific to the mongodb setup?